src/components/CoinList/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { gql, useQuery } from "@apollo/client";
+import CoinCard from "../CoinCard";
+import Loading from "../Loading";
+
+const FETCH_COIN_LIST = gql`
+  query FetchCoinsList {
+    coinsList
+      @rest(type: "ListPayload", path: "/data/top/totalvolfull?tsym=USD") {
+      Data @type(name: "DataPayload") {
+        CoinInfo @type(name: "CoinInfoPayload") {
+          Id
+          Name
+          FullName
+          ImageUrl
+        }
+        DISPLAY @type(name: "DisplayPayload") {
+          USD @type(name: "USDPayLoad") {
+            PRICE
+            OPENDAY
+            HIGHDAY
+            LOWDAY
+            OPEN24HOUR
+          }
+        }
+      }
+    }
+  }
+`;
+const CoinList = () => {
+  const { loading, error, data } = useQuery(FETCH_COIN_LIST);
+  return (
+    <>
+      {loading ? (
+        <div className="m-24">
+          <Loading />
+        </div>
+      ) : (
+        data.coinsList.Data.map((data: any) => (
+          <CoinCard
+            key={data.CoinInfo.Id}
+            coinInfo={data.CoinInfo}
+            price={data.DISPLAY.USD.PRICE}
```

Comment:
avoid more than 1 dot accessor, the list component shouldn't be burdened with accessing price inside data, you can do this inside the coin card component.